src/transformers/tokenization_utils_fast.py

```diff
@@ -172,10 +172,12 @@ def __init__(self, *args, **kwargs):
         # allows converting a slow -> fast, non-legacy: if the `tokenizer.json` does not have all the added tokens
         # uses the information stored in `added_tokens_decoder`.
         # this is costly for fast tokenizers as we re-compute the regex again. But not all tokens are added tokens
+        # Use hash to speed up the very slow operation `token not in added_tokens_decoder`.
+        added_tokens_decoder_hash = {hash(token) for token in self.added_tokens_decoder}
```

Comment:
```suggestion
        added_tokens_decoder_hash = {hash(token.__str__()) for token in self.added_tokens_decoder}
```
would that be even faster? hash the string rep?

Reply:
it's on par :-) OK for me to both, but if using str, I would just do it `str(token)`

Reply:
No, you need to make sure normalized, left, right and single word is not the same

Reply:
`str` and `__str__` will give the same thing. Do you mean to use `__repr__`?

That will give
> 'AddedToken("<s>", rstrip=False, lstrip=False, single_word=False, normalized=False, special=True)'
while `str` and `__str__`
will give, for example,

> </s>

note `str` will call `__str__` under the hood.

Reply:
yeah sorry repr